### PR TITLE
Refactors "strpos" calls in /apps/workflowengine

### PR DESCRIPTION
**Pull request by fsamapoor**
## Summary
Following https://github.com/nextcloud/server/pull/38261 and https://github.com/nextcloud/server/pull/38260, I have replaced `strpos` calls in `/apps/workflowengine` namespace as well.

### CODE REVIEW

**Review by come-nc — APPROVED** [review]

**Review by solracsf — APPROVED** [review]

File: apps/workflowengine/lib/Check/FileMimeType.php [hunk 1]
```diff
@@ -157,11 +157,11 @@
 	protected function isWebDAVRequest() {
 		return substr($this->request->getScriptName(), 0 - strlen('/remote.php')) === '/remote.php' && (
 			$this->request->getPathInfo() === '/webdav' ||
-			strpos($this->request->getPathInfo(), '/webdav/') === 0 ||
+			str_starts_with($this->request->getPathInfo(), '/webdav/') ||
```
[the remark this answers is not in the document]
Comment (come-nc) [reply apps/workflowengine/lib/Check/FileMimeType.php:160]: @fsamapoor Could you add the `?? ''` needed to make psalm happy about this PR?

[the remark this answers is not in the document]
Comment (fsamapoor) [reply apps/workflowengine/lib/Check/FileMimeType.php]: Thank you for pointing these out.

**Comment by fsamapoor** [comment]
Hello @come-nc, Sorry to bother you.
I understand that reviewing PRs can take some time, but sometimes my PRs do not get merged after being approved and some of them have more than two approvals. Should I merge them myself once all the required checks have passed? Are there any considerations in place to merge PRs?

Thank you for your assistance.

**Comment by fsamapoor** [comment]
> > Hello @come-nc,
> > Sorry to bother you. I understand that reviewing PRs can take some time, but sometimes my PRs do not get merged after being approved and some of them have more than two approvals. Should I merge them myself once all the required checks have passed? Are there any considerations in place to merge PRs?
> > Thank you for your assistance.
> 
> Hey, do not hesitate to notify us if it goes unnoticed for too long. For PRs on master at least.
> 
> (But according to psalm on this one ?? is used where ?: should have been since getPathInfo can return false. But then ?: I suppose would eat strings like '0'. Not sure what the clean thing is, also I did not check what getPathInfo actually returns.)

Sure. Thanks.

I applied these changes based on your suggestion in the review:
https://github.com/nextcloud/server/pull/38604#discussion_r1250485079


I just checked the codebase. The `getPathInfo` method in the `Request` class returns `string`, but the `IRequest` interface has the `string|false` annotation for the `getPathInfo` method. So either way, the code should work fine with or without any of the `??` or `?:` operators. Making Psalm happy would need to update the  `getPathInfo` method annotations in the `Request` class. Let me know if I should revert the commit that adds the `??` operator to the mix.

**Comment by come-nc** [comment]
> I applied these changes based on your suggestion in the review: [#38604 (comment)](https://github.com/nextcloud/server/pull/38604#discussion_r1250485079)
> 
> I just checked the codebase. The `getPathInfo` method in the `Request` class returns `string`, but the `IRequest` interface has the `string|false` annotation for the `getPathInfo` method. So either way, the code should work fine with or without any of the `??` or `?:` operators. Making Psalm happy would need to update the `getPathInfo` method annotations in the `Request` class. Let me know if I should revert the commit that adds the `??` operator to the mix.

Let’s leave it like that for now.

**Pull request merged**